wgpu-core/src/device/mod.rs

```diff
@@ -1478,7 +1480,51 @@ impl<G: GlobalIdentityHandlerFactory> Global<G> {
         let device = &device_guard[texture.device_id.value];
 
         let view_kind = match desc.dimension {
-            Some(dim) => conv::map_texture_view_dimension(dim),
+            Some(dim) => {
+                use hal::image::Kind;
+
+                let valid_view_dim = match texture.dimension {
+                    TextureDimension::D1 => matches!(dim, TextureViewDimension::D1),
+                    TextureDimension::D2 => matches!(
+                        dim,
+                        TextureViewDimension::D2
+                            | TextureViewDimension::D2Array
+                            | TextureViewDimension::Cube
+                            | TextureViewDimension::CubeArray
+                    ),
+                    TextureDimension::D3 => matches!(dim, TextureViewDimension::D3),
+                };
+
+                if !valid_view_dim {
+                    return Err(
+                        resource::CreateTextureViewError::InvalidTextureViewDimension {
+                            view: dim,
+                            image: texture.dimension,
+                        },
+                    );
+                }
+
+                match (dim, texture.kind) {
```

Comment:
this could be rewritten somewhat cleaner as:
```rust
if let Kind::D2(_, _, depth, _) = texture.kind {
    match dim {
      TextureViewDimension::Cube if depth != 6 => return Err(...),
      TextureViewDimension::CubeArray if depth % 6 != 0 => return Err(...),
      _ => {}
    }
}
```